.github/workflows/upload.yaml

```diff
@@ -0,0 +1,77 @@
+name: Upload to website
+
+on:
+  pull_request:
+    branches:
+      - main
+
+  workflow_dispatch:
+
+jobs:
+  Upload-staging:
+    if: ${{ github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true }}
+    name: Upload to staging website bucket
+    environment:
+      name: staging
+      url: 'https://staging.pypi.opentrons.com'
+    runs-on: ubuntu-latest 
+    steps:
+      - name: Checkout
+        uses; actions/checkout@v2
+
+      - name: Upload
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read --follow-symlinks --delete --exclude '.git*/*' --exclude '.github/*'
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+          SOURCE_DIR: ''
```

Comment:
We should avoid passing in AWS credentials like this.
Let us use[ `configure-aws-credentials`](https://github.com/aws-actions/configure-aws-credentials)
This means that along with the user, we will need to setup the the [GitHub Identity Provider](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role)
Looks more like the below, except I would like use to use the GitHub environments fr the repository feature to hold the  role arn probably.
```yaml
      - name: Set AWS role if prod.
        if: (github.event.inputs.env == 'prod') == true
        run: |
          echo "ROLE_FOR_CREDENTIALS=arn:aws:iam::ROLE" >> $GITHUB_ENV
          echo "ENVIRONMENT=prod" >> $GITHUB_ENV
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          role-to-assume: ${{ env.ROLE_FOR_CREDENTIALS }}
          aws-region: us-east-2
```

Reply:
This has also been adjusted!